### ISSUE TITLE
LevelUp instead of path

### ISSUE THREAD

**Issue by andrewdeandrade** [dominictarr/content-addressable-store#1]
Just came across this module that looks like it perfectly encapsulates a feature I already wrote in another module I'm working on. Would you accept a pull request that allows this module to take LevelDB as the first argument instead of a path? (but still supports a path for legacy sake)

Also, is there a reason you don't already support LevelUp for this? I actually assumed that is how it would work internally since you already do so much with LevelUp?

**Comment by dominictarr** [comment]
This module just uses files, and doesn't use levelup at all! (it's modeled on how git stores blobs)

Although, I have a level version of this module, but it's embedded into another module.

I should refactor it out. https://github.com/dominictarr/level-content-cache/blob/master/level-content-addressable-store.js

I only left it in there because I didn't have any thing I needed the level version for.

**Comment by andrewdeandrade** [comment]
Yeah, I found that level version after I posted this issue. Ideally I want a content addressable store that also works in the browser in the future, which is why I wanted something leveldb based. My plan is to abstract away filesystem access so that whatever is in the CAS can either read from disk or via ajax request.

**Comment by dominictarr** [comment]
oh sure - well as long as you have the same API is all circumstances, that is the important thing. The api is pretty small, so it makes that easy.

**Comment by dominictarr** [comment]
okay I pulled out and published level-content-addressable-store
